actionpack/test/controller/base_test.rb

```diff
@@ -168,13 +168,13 @@ def test_process_should_be_precise
     assert_equal "The action 'non_existent' could not be found for EmptyController", exception.message
   end
 
-  if defined?(DidYouMean) && DidYouMean.respond_to?(:correct_error)
+  if defined?(DidYouMean::SpellChecker)
     def test_exceptions_have_suggestions_for_fix
       use_controller SimpleController
       exception = assert_raise AbstractController::ActionNotFound do
-        get :non_existent
+        get :hlelo
       end
-      assert_match "Did you mean?", exception.message
+      assert_match "Did you mean?  hlelo", exception.message
```

Comment:
This suggestion also doesn't seem accurate. Something to look into later or even as part of this PR.